pkg/token/webbrowser.go

```diff
@@ -0,0 +1,76 @@
+package token
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/go-autorest/autorest/adal"
+)
+
+type WebBrowserToken struct {
+	clientID    string
+	resourceID  string
+	tenantID    string
+	oAuthConfig adal.OAuthConfig
+}
+
+// newAzureCLIToken returns a TokenProvider that will fetch a token for the user currently logged into the Azure CLI.
+// Required arguments include an oAuthConfiguration object and the resourceID (which is used as the scope)
+func newWebBrowserTokenProvider(oAuthConfig adal.OAuthConfig, clientID, resourceID, tenantID string) (TokenProvider, error) {
+	if clientID == "" {
+		return nil, errors.New("clientID cannot be empty")
+	}
+	if resourceID == "" {
+		return nil, errors.New("resourceID cannot be empty")
+	}
+	if tenantID == "" {
+		return nil, errors.New("tenantID cannot be empty")
+	}
+
+	return &WebBrowserToken{
+		clientID:    clientID,
+		resourceID:  resourceID,
+		tenantID:    tenantID,
+		oAuthConfig: oAuthConfig,
+	}, nil
+}
+
+// Token fetches an azcore.AccessToken from the Azure CLI SDK and converts it to an adal.Token for use with kubelogin.
+func (p *WebBrowserToken) Token() (adal.Token, error) {
+	emptyToken := adal.Token{}
+
+	// Request a new Azure CLI token provider
+	cred, err := azidentity.NewInteractiveBrowserCredential(&azidentity.InteractiveBrowserCredentialOptions{
```

Comment:
you actually need to configure Authority so that it would work in other clouds.
Example: https://github.com/denisenkom/go-mssqldb/blob/master/azuread/configuration.go#L150

Reply:
let's configure the authority in newWebBrowerTokenProvider

Reply:
we need one additional flags:
* `--cloud` so that you can setup the `azcore.ClientOptions.Cloud`. [ref](https://github.com/Azure/azure-sdk-for-go/blob/c5358ab6c0bf824403f0ecdbd164c882d794793d/sdk/azidentity/MIGRATION.md#azidentity)

besides, we need to take `--authority-host` into account:
* when `--authority-host` is set, we will use it instead of setting `azcore.ClientOptions.Cloud`
* probably should do a validation to check if `--authority-host` and the computed authority host from `--cloud` are the same when both are set

Reply:
Done in the most recent commit f8e875a. Please check it out. Thanks!